test/test_image.py

```diff
@@ -54,176 +53,202 @@ def normalize_dimensions(img_pil):
-            encode_png(torch.empty((5, 100, 100), dtype=torch.uint8))
-
-    def test_write_png(self):
-        with get_tmp_dir() as d:
-            for img_path in get_images(IMAGE_DIR, '.png'):
-                pil_image = Image.open(img_path)
-                img_pil = torch.from_numpy(np.array(pil_image))
-                img_pil = img_pil.permute(2, 0, 1)
-
-                filename, _ = os.path.splitext(os.path.basename(img_path))
-                torch_png = os.path.join(d, '{0}_torch.png'.format(filename))
-                write_png(img_pil, torch_png, compression_level=6)
-                saved_image = torch.from_numpy(np.array(Image.open(torch_png)))
-                saved_image = saved_image.permute(2, 0, 1)
-
-                assert_equal(img_pil, saved_image)
-
-    def test_read_file(self):
-        with get_tmp_dir() as d:
-            fname, content = 'test1.bin', b'TorchVision\211\n'
-            fpath = os.path.join(d, fname)
-            with open(fpath, 'wb') as f:
-                f.write(content)
-
-            data = read_file(fpath)
-            expected = torch.tensor(list(content), dtype=torch.uint8)
-            assert_equal(data, expected)
-            os.unlink(fpath)
-
-        with self.assertRaisesRegex(
-                RuntimeError, "No such file or directory: 'tst'"):
-            read_file('tst')
-
-    def test_read_file_non_ascii(self):
-        with get_tmp_dir() as d:
-            fname, content = '日本語(Japanese).bin', b'TorchVision\211\n'
-            fpath = os.path.join(d, fname)
-            with open(fpath, 'wb') as f:
-                f.write(content)
-
-            data = read_file(fpath)
-            expected = torch.tensor(list(content), dtype=torch.uint8)
-            assert_equal(data, expected)
-            os.unlink(fpath)
-
-    def test_write_file(self):
-        with get_tmp_dir() as d:
-            fname, content = 'test1.bin', b'TorchVision\211\n'
-            fpath = os.path.join(d, fname)
-            content_tensor = torch.tensor(list(content), dtype=torch.uint8)
-            write_file(fpath, content_tensor)
-
-            with open(fpath, 'rb') as f:
-                saved_content = f.read()
-            self.assertEqual(content, saved_content)
-            os.unlink(fpath)
-
-    def test_write_file_non_ascii(self):
-        with get_tmp_dir() as d:
-            fname, content = '日本語(Japanese).bin', b'TorchVision\211\n'
-            fpath = os.path.join(d, fname)
-            content_tensor = torch.tensor(list(content), dtype=torch.uint8)
-            write_file(fpath, content_tensor)
-
-            with open(fpath, 'rb') as f:
-                saved_content = f.read()
-            self.assertEqual(content, saved_content)
-            os.unlink(fpath)
+@pytest.mark.parametrize('img_path', [
+    pytest.param(jpeg_path, id=_get_safe_image_name(jpeg_path))
+    for jpeg_path in get_images(IMAGE_ROOT, ".jpg")
+])
+@pytest.mark.parametrize('conversion', [
```

Comment:
this is totally fine as-is so no need to change anything, but in case you don't know, we can parametrize tuples as well:

```py
@pytest.mark.parametrize('pil_mode, mode', [(None, ImageReadMode.UNCHANGED), ...])
def test_decode_jpeg(img_path, pil_mode, mode):
```

So we don't need to unpack in the code.

Reply:
Done!